create_fork.sh

```diff
@@ -0,0 +1,77 @@
+#!/bin/bash
+#
+#  Copyright 2023 The original authors
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+
+set -eo pipefail
+
+SOURCE_FORK="baseline"
+
+usage() {
+  echo "Usage: create_fork.sh [-s <source fork>] <fork name>"
+  echo "  -s <source fork>  The name of the fork to copy from (default: baseline)"
+  echo "  <fork name>       The name of the fork to create"
+  exit 1
+}
+
+# Parse
+while getopts ":s:" opt; do
+  case ${opt} in
+    s )
+      SOURCE_FORK=$OPTARG
+      ;;
+    \? )
+      usage
+      exit 1
+      ;;
+    : )
+      echo "Invalid option: $OPTARG requires an argument" 1>&2
+      exit 1
+      ;;
+  esac
+done
+
+FORK=${@:$OPTIND:1}
+if [ -z "$FORK" ]
+  then
+    usage
+    exit 1
+fi
+
+# helper function
+function substitute_in_file {
+  if [[ "$OSTYPE" == "darwin"* ]]; then
+    sed -i '' "s/$1/$2/g" $3
+  else
+    sed -i "s/$1/$2/g" $3
+  fi
+}
+
+set -x
+
+# create new fork
+cp prepare_$SOURCE_FORK.sh prepare_$FORK.sh
+
+cp calculate_average_$SOURCE_FORK.sh calculate_average_$FORK.sh
+substitute_in_file $SOURCE_FORK $FORK calculate_average_$FORK.sh
+
+if [ $SOURCE_FORK == "baseline" ]; then
+  cp src/main/java/dev/morling/onebrc/CalculateAverage.java src/main/java/dev/morling/onebrc/CalculateAverage_$FORK.java
```

Comment:
FWIW not every valid github username gives a valid java class name, like mine for instance.

Reply:
"Falsehoods Programmers Believe About Names" 😂

Reply:
We could validate the fork name has only `[a-zA-Z0-9_]` and then error otherwise to let the user fix, I don't want to prescribe a naming convention in this script that entrants that don't use the script may or may not follow

Reply:
I don't think it's worth the effort really. What you have should work 99% of the time, what more to ask for.

Reply:
No effort required, the AI crushed this task 😉 

<img width="865" alt="image" src="https://github.com/gunnarmorling/1brc/assets/91577/fb6f1b46-0f7a-4d8c-bf9d-fa29a810015f">